graphql_compiler/global_utils.py

```diff
@@ -59,3 +59,22 @@ def is_same_type(left, right):
         return is_same_type(left.of_type, right.of_type)
     else:
         return False
+
+
+def validate_that_mappings_have_the_same_keys(
+    mapping1: Mapping[str, Any], mapping2: Mapping[str, Any]
+) -> None:
+    """Validate that the mappings have the same keys."""
+    mapping1_keys = set(mapping1.keys())
+    mapping2_keys = set(mapping2.keys())
+
+    diff1 = mapping1_keys.difference(mapping2_keys)
+    diff2 = mapping2_keys.difference(mapping1_keys)
+
+    if diff1 or diff2:
+        error_message_list = ["Expected mappings to have the same keys."]
+        if diff1:
+            error_message_list.append(f" Keys in the first mapping but not the second: {diff1}.")
+        if diff2:
+            error_message_list.append(f" Keys in the second mapping but not the first: {diff2}.")
+        raise AssertionError(" ".join(error_message_list))
```

Comment:
I think you'll end up with double spacing here since you're joining with a `" "` and started your error messages with a space as well.